release: ship libgeos.* and libgeos_c.* files

otan opened the pull request:
This commit adds libgeos.* and libgeos_c.* files to be uploaded to S3 at
`s3://cockroach/cockroach/lib/` as well as adding the files into the
.tar.gz and .zip archives.

Release note (general change): The CockroachDB tarballs now include
libgeos and libgeos_c static libraries, which are required to be copied
to /usr/local/lib/cockroach for UNIX/MacOS systems to perform certain
geospatial operations. The Windows location will be finalized at a later
date. This location is configurable with the `--geo-libs` flag.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/51108)
<!-- Reviewable:end -->

otan committed:
release: ship libgeos.* and libgeos_c.* files

This commit adds libgeos.* and libgeos_c.* files to be uploaded to S3 at
`s3://cockroach/cockroach/lib/` as well as adding the files into the
.tar.gz and .zip archives.

Release note (general change): The CockroachDB tarballs now include
libgeos and libgeos_c static libraries, which are required to be copied
to /usr/local/lib/cockroach for UNIX/MacOS systems to perform certain
geospatial operations. The Windows location will be finalized at a later
date. This location is configurable with the `--geo-libs` flag.

jlinder reviewed (approved):
LGTM

otan commented:
bors r=jlinder

tftr!

otan commented:
looks like a flake...

bors r=jlinder
